### DISCUSSION TITLE
Rename pull request as "merge request" 🤪

### DISCUSSION THREAD

**Discussion by NitinIsFit** [community/community#27561] [Pull Requests] [answered]
Pull request is a confusing word and most people don't understand this concept, please change `pull request` into `merge request` or `code merge request`

**Comment by radar-blip** [comment, marked as answer]
I agree, merge request would be better.

**Comment by jfullstackdev** [comment]
I'm very sure they decided to call it as such, 

GitLab and GitHub were both based on Git, Git has `merge` and `pull`, seems simple right ? but it's not 

see this 

https://stackoverflow.com/questions/21756614/difference-between-git-merge-origin-master-and-git-pull

**Reply by NitinIsFit** [reply]
> GitHub have both `merge` and `pull`

Well, merge is simply merge and pull request can be merge request right?

**Reply by YevheniiPokhvalii** [reply]
Pull request is a very confusing term for the newcomers. People that see it the first time do not care what happens under the hood. You just throw a request to merge your code and merge request is a general term outside Gitlab. It is only Github that uses pull requests.

**Reply by NitinIsFit** [reply]
> Pull request is a very confusing term for the newcomers. People that see it the first time do not care what happens under the hood. You just throw a request to merge your code and merge request is a general term outside Gitlab. It is only Github that uses pull requests.

Pull request is actually a merge request, because it's a request to merge the files

**Comment by oe3gwu** [comment]
I agree.

**Comment by 0x501D** [comment]
`merge request` is confusing. 
I would gladly rename it to pull in Gitlab if there was such an opportunity

**Reply by YevheniiPokhvalii** [reply]
This "pull request" naming convention is confusing for everyone, except old programmers. It's like sort of a legal lexicon that was relevant back in the day but became outdated and still used.
It does not even sound right to me till this date and I'm not the only one: 
https://stackoverflow.com/questions/21657430/why-is-a-git-pull-request-not-called-a-push-request

As a user, I consider it a "push" and not "pull", I don't care that it's a "pull" from the target repository perspective. So "merge request" sounds at least neutral.

**Comment by mindplay-dk** [comment]
It's 2026, please can we change it to "morg request" 🤪

**Comment by typed-sigterm** [comment]
the unfixed bug already turned out as a feature 🤪

**Comment by hammy33897** [comment]
Still they call it pull request. Stop the madness!
Can we create a petition to get this changed, it is soooo... confusing and inconsistent with other version control tools.

**Reply by 0x501D** [reply]
I propose an opposition: put together a petition in GitLab to rename it to Pull there.
